services/src/main/java/org/keycloak/services/resources/account/AccountFormService.java

```diff
@@ -585,7 +585,7 @@ public Response processPasswordUpdate(final MultivaluedMap<String, String> formD
             ServicesLogger.LOGGER.failedToUpdatePassword(me);
             setReferrerOnPage();
             errorEvent.detail(Details.REASON, me.getMessage()).error(Errors.PASSWORD_REJECTED);
-            return account.setError(Response.Status.INTERNAL_SERVER_ERROR, me.getMessage(), me.getParameters()).createResponse(AccountPages.PASSWORD);
+            return account.setError(Response.Status.OK, me.getMessage(), me.getParameters()).createResponse(AccountPages.PASSWORD);
```

Comment:
I am not sure about status OK, could it be changed to BAD_REQUEST similarly like in the previous block of read-only error?

Reply:
Hi, Sounds good, as that could be handle better than a 500. 

Is it a bad request if the validation policy is wrong. I was assuming the request itself was ok, but just that the policy validation didn't match the rules

Happy to update the PR to BAD_REQUEST once I test it locally

Reply:
I have tested locally setting it to BAD_REQUEST however this doesn't return the response as expected. It returns the html unrendered version which looks odd

> <!doctype html> <html> <head> ....

Reply:
Hi @mposolda didn't hear anything back. I tried changing the value you recommended but it didn't work. The response was unrendered suggesting that the bad request isn't handled by the ui.

Reply:
@farmersmurf Could you try to test this with Keycloak on the wildfly server? It seems this may be broken on KC on embedded undertow, but it seem to work on Wildfly.

Reply:
Aha, ok, Will do, thanks

Reply:
BTV. There is the issue https://issues.jboss.org/browse/KEYCLOAK-11000 about the bad response seeing unrendered HTML. However when this bug is fixed, we should return 400 rather than 200 IMO.

Reply:
Cool, interesting there was a rest easy bug from wildfly. I'll make some time to test locally. Cheers